qiskit_experiments/experiment_data.py

```diff
@@ -89,6 +90,13 @@ def experiment_id(self) -> str:
         """Return the experiment id."""
         return self._id
 
+    def metadata(self) -> Dict:
```

Comment:
Is there any mechanism that maps metadata to the experiment data entry? I'm not clear how we can leverage this experiment metadata. Because this seems to return a nested dict of `Dict[job_id: str, Dict[str, Any]]`, it seems hard to access to the metadata values. We may need something like
```python
num_qubits = set()
for job_metadata in metadata.values():
    num_qubits.add(job_metadata['num_qubits'])
if len(num_qubits) == 1:
    num_qubits = list(num_qubits)[0]
else:
   raise AppropriateError()
```
to access to the `num_qubits`. I don't think this is quite handy though data structure is reasonable.

Reply:
Yeah I think I need to change the job metadata subfield needs to be a list, rather than a dict, so you can know the order of entries and in principle from this could deduce which result values correspond to each job and set of option values.